core/src/main/scala/org/http4s/Method.scala

```diff
@@ -44,8 +44,8 @@ object Method {
   }
 
   // Type tags for a method allowing a body or not
-  sealed trait PermitsBody extends Method
-  sealed trait NoBody extends Method
+  sealed trait PermitsBody
```

Comment:
Would dotty allow these to take a self type of `Method`?  They shouldn't be mixed into anything else.

```scala
sealed trait PermitsBody { self: Method => }
```

If not, this is fine.

Reply:
let me try...
but this is sealed so only types in this file can extend it, is it really necessary?

Reply:
it works

Reply:
Yeah, arguably not necessary since it's sealed, but more intention revealing.  I'd prefer it, but am fine without it if it causes problems.  Dotty support is more important.